Tools/clinic/clinic.py

```diff
@@ -2273,15 +2276,16 @@ def parse(self, input: str) -> str:
                             os.makedirs(dirname)
                         except FileExistsError:
                             if not os.path.isdir(dirname):
-                                fail("Can't write to destination {}, "
-                                     "can't make directory {}!".format(
-                                        destination.filename, dirname))
+                                fail(f"Can't write to destination "
+                                     f"{destination.filename!r}, "
```

Comment:
This would be better grammar:

```suggestion
                                     f"{destination.filename!r}; "
```